// primeng/components/tabview/tabview.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/tabview/tabview';
import * as i2 from '@angular/common';
import * as i3 from 'primeng/components/common/shared';
import * as i4 from '../common/shared.ngfactory';
export const TabViewModuleNgFactory:i0.NgModuleFactory<i1.TabViewModule> = i0.ɵcmf(i1.TabViewModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i3.SharedModule,i3.SharedModule,
          ([] as any[])),i0.ɵmpd(512,i1.TabViewModule,i1.TabViewModule,([] as any[]))]);
    });
const styles_TabView:any[] = ([] as any[]);
export const RenderType_TabView:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_TabView,
    data:{}});
function View_TabView_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ul',[['p-tabViewNav',
      ''],['role','tablist']],[[2,'ui-tabview-nav',(null as any)],[2,'ui-helper-reset',
      (null as any)],[2,'ui-helper-clearfix',(null as any)],[2,'ui-widget-header',
      (null as any)],[2,'ui-corner-all',(null as any)]],[[(null as any),'onTabClick'],
      [(null as any),'onTabCloseClick']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('onTabClick' === en)) {
      const pd_0:any = ((<any>_co.open($event.originalEvent,$event.tab)) !== false);
      ad = (pd_0 && ad);
    }
    if (('onTabCloseClick' === en)) {
      const pd_1:any = ((<any>_co.close($event.originalEvent,$event.tab)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },View_TabViewNav_0,RenderType_TabViewNav)),i0.ɵdid(49152,(null as any),0,i1.TabViewNav,
      ([] as any[]),{tabs:[0,'tabs'],orientation:[1,'orientation']},{onTabClick:'onTabClick',
          onTabCloseClick:'onTabCloseClick'})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_5:any = _co.tabs;
    const currVal_6:any = _co.orientation;
    _ck(_v,1,0,currVal_5,currVal_6);
  },(_ck,_v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    const currVal_2:any = true;
    const currVal_3:any = true;
    const currVal_4:any = true;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_TabView_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'ul',[['p-tabViewNav',
      ''],['role','tablist']],[[2,'ui-tabview-nav',(null as any)],[2,'ui-helper-reset',
      (null as any)],[2,'ui-helper-clearfix',(null as any)],[2,'ui-widget-header',
      (null as any)],[2,'ui-corner-all',(null as any)]],[[(null as any),'onTabClick'],
      [(null as any),'onTabCloseClick']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('onTabClick' === en)) {
      const pd_0:any = ((<any>_co.open($event.originalEvent,$event.tab)) !== false);
      ad = (pd_0 && ad);
    }
    if (('onTabCloseClick' === en)) {
      const pd_1:any = ((<any>_co.close($event.originalEvent,$event.tab)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },View_TabViewNav_0,RenderType_TabViewNav)),i0.ɵdid(49152,(null as any),0,i1.TabViewNav,
      ([] as any[]),{tabs:[0,'tabs'],orientation:[1,'orientation']},{onTabClick:'onTabClick',
          onTabCloseClick:'onTabCloseClick'})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_5:any = _co.tabs;
    const currVal_6:any = _co.orientation;
    _ck(_v,1,0,currVal_5,currVal_6);
  },(_ck,_v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    const currVal_2:any = true;
    const currVal_3:any = true;
    const currVal_4:any = true;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export function View_TabView_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),14,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵdid(278528,(null as any),
      0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_TabView_1)),i0.ɵdid(16384,(null as any),
      0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),3,'div',[['class','ui-tabview-panels']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                '])),i0.ɵncd((null as any),0),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TabView_2)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.TabView = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = ('ui-tabview ui-widget ui-widget-content ui-corner-all ui-tabview-' + _co.orientation);
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,3,0,currVal_2);
    const currVal_3:any = (_co.orientation != 'bottom');
    _ck(_v,6,0,currVal_3);
    const currVal_4:any = (_co.orientation == 'bottom');
    _ck(_v,14,0,currVal_4);
  },(null as any));
}
export function View_TabView_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'p-tabView',([] as any[]),
      (null as any),(null as any),(null as any),View_TabView_0,RenderType_TabView)),
      i0.ɵdid(1097728,(null as any),1,i1.TabView,[i0.ElementRef],(null as any),(null as any)),
      i0.ɵqud(603979776,1,{tabPanels:1})],(null as any),(null as any));
}
export const TabViewNgFactory:i0.ComponentFactory<i1.TabView> = i0.ɵccf('p-tabView',
    i1.TabView,View_TabView_Host_0,{orientation:'orientation',style:'style',styleClass:'styleClass',
        controlClose:'controlClose',lazy:'lazy',activeIndex:'activeIndex'},{onChange:'onChange',
        onClose:'onClose'},['*']);
const styles_TabPanel:any[] = ([] as any[]);
export const RenderType_TabPanel:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_TabPanel,
    data:{}});
function View_TabPanel_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,'p-templateLoader',
      ([] as any[]),(null as any),(null as any),(null as any),i4.View_TemplateLoader_0,
      i4.RenderType_TemplateLoader)),i0.ɵdid(245760,(null as any),0,i3.TemplateLoader,
      [i0.ViewContainerRef],{template:[0,'template']},(null as any)),(_l()(),i0.ɵand(0,
      (null as any),(null as any),0))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.contentTemplate;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_TabPanel_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'div',[['class',
      'ui-tabview-panel ui-widget-content'],['role','tabpanel']],[[4,'display',(null as any)],
      [1,'aria-hidden',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),i0.ɵncd((null as any),0),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_TabPanel_2)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = (_co.contentTemplate && (_co.cache? _co.loaded: _co.selected));
    _ck(_v,5,0,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.selected? 'block': 'none');
    const currVal_1:boolean = !_co.selected;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
export function View_TabPanel_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_TabPanel_1)),i0.ɵdid(16384,
      (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.TabPanel = _v.component;
    const currVal_0:boolean = !_co.closed;
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_TabPanel_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),2,'p-tabPanel',
      ([] as any[]),(null as any),(null as any),(null as any),View_TabPanel_0,RenderType_TabPanel)),
      i0.ɵdid(1228800,(null as any),1,i1.TabPanel,[i0.ViewContainerRef],(null as any),
          (null as any)),i0.ɵqud(603979776,1,{templates:1})],(null as any),(null as any));
}
export const TabPanelNgFactory:i0.ComponentFactory<i1.TabPanel> = i0.ɵccf('p-tabPanel',
    i1.TabPanel,View_TabPanel_Host_0,{header:'header',disabled:'disabled',closable:'closable',
        headerStyle:'headerStyle',headerStyleClass:'headerStyleClass',leftIcon:'leftIcon',
        rightIcon:'rightIcon',cache:'cache',selected:'selected'},{},['*']);
const styles_TabViewNav:any[] = ([] as any[]);
export const RenderType_TabViewNav:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_TabViewNav,
    data:{}});
function View_TabViewNav_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'ui-tabview-left-icon fa']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any))],(_ck,_v) => {
    const currVal_0:any = 'ui-tabview-left-icon fa';
    const currVal_1:any = (<any>(<any>_v.parent).parent).context.$implicit.leftIcon;
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_TabViewNav_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'ui-tabview-right-icon fa']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any))],(_ck,_v) => {
    const currVal_0:any = 'ui-tabview-right-icon fa';
    const currVal_1:any = (<any>(<any>_v.parent).parent).context.$implicit.rightIcon;
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_TabViewNav_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class',
      'ui-tabview-close fa fa-close']],(null as any),[[(null as any),'click']],(_v,
      en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.clickClose($event,(<any>(<any>_v.parent).parent).context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
function View_TabViewNav_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),19,'li',[['role',
      'tab']],[[1,'aria-expanded',0],[1,'aria-selected',0]],[[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.clickTab($event,(<any>_v.parent).context.$implicit)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-tabview-selected ui-state-active':0,
      'ui-state-disabled':1}),i0.ɵdid(278528,(null as any),0,i2.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n                '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'a',
      [['href','#']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_TabViewNav_3)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class','ui-tabview-title']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['',''])),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TabViewNav_4)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_TabViewNav_5)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n            ']))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_2:any = _co.getDefaultHeaderClass((<any>_v.parent).context.$implicit);
    const currVal_3:any = _ck(_v,2,0,(<any>_v.parent).context.$implicit.selected,(<any>_v.parent).context.$implicit.disabled);
    _ck(_v,1,0,currVal_2,currVal_3);
    const currVal_4:any = (<any>_v.parent).context.$implicit.headerStyle;
    _ck(_v,3,0,currVal_4);
    const currVal_5:any = (<any>_v.parent).context.$implicit.leftIcon;
    _ck(_v,8,0,currVal_5);
    const currVal_7:any = (<any>_v.parent).context.$implicit.rightIcon;
    _ck(_v,14,0,currVal_7);
    const currVal_8:any = (<any>_v.parent).context.$implicit.closable;
    _ck(_v,18,0,currVal_8);
  },(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.selected;
    const currVal_1:any = (<any>_v.parent).context.$implicit.selected;
    _ck(_v,0,0,currVal_0,currVal_1);
    const currVal_6:any = (<any>_v.parent).context.$implicit.header;
    _ck(_v,11,0,currVal_6);
  });
}
function View_TabViewNav_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_TabViewNav_2)),i0.ɵdid(16384,
      (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    const currVal_0:boolean = !_v.context.$implicit.closed;
    _ck(_v,2,0,currVal_0);
  },(null as any));
}
export function View_TabViewNav_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_TabViewNav_1)),i0.ɵdid(802816,
      (null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
      {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],
      (_ck,_v) => {
        var _co:i1.TabViewNav = _v.component;
        const currVal_0:any = _co.tabs;
        _ck(_v,2,0,currVal_0);
      },(null as any));
}
export function View_TabViewNav_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['p-tabViewNav',
      '']],[[2,'ui-tabview-nav',(null as any)],[2,'ui-helper-reset',(null as any)],
      [2,'ui-helper-clearfix',(null as any)],[2,'ui-widget-header',(null as any)],
      [2,'ui-corner-all',(null as any)]],(null as any),(null as any),View_TabViewNav_0,
      RenderType_TabViewNav)),i0.ɵdid(49152,(null as any),0,i1.TabViewNav,([] as any[]),
      (null as any),(null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = true;
    const currVal_1:any = true;
    const currVal_2:any = true;
    const currVal_3:any = true;
    const currVal_4:any = true;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
export const TabViewNavNgFactory:i0.ComponentFactory<i1.TabViewNav> = i0.ɵccf('[p-tabViewNav]',
    i1.TabViewNav,View_TabViewNav_Host_0,{tabs:'tabs',orientation:'orientation'},{onTabClick:'onTabClick',
        onTabCloseClick:'onTabCloseClick'},([] as any[]));
