// primeng/components/editor/editor.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/editor/editor';
import * as i2 from '@angular/common';
import * as i3 from 'primeng/components/common/shared';
import * as i4 from '@angular/forms';
import * as i5 from 'primeng/components/dom/domhandler';
export const EditorModuleNgFactory:i0.NgModuleFactory<i1.EditorModule> = i0.ɵcmf(i1.EditorModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i3.SharedModule,i3.SharedModule,
          ([] as any[])),i0.ɵmpd(512,i1.EditorModule,i1.EditorModule,([] as any[]))]);
    });
const styles_Editor:any[] = ([] as any[]);
export const RenderType_Editor:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Editor,
    data:{}});
function View_Editor_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class',
      'ui-editor-toolbar ui-widget-header ui-corner-top']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),
      i0.ɵncd((null as any),0),(_l()(),i0.ɵted((null as any),['\n            ']))],
      (null as any),(null as any));
}
function View_Editor_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),76,'div',[['class',
      'ui-editor-toolbar ui-widget-header ui-corner-top']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),25,'span',[['class','ql-formats']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),10,'select',[['class','ql-header']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          1,'option',[['value','1']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['Heading'])),(_l()(),i0.ɵted((null as any),
          ['\n                      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          1,'option',[['value','2']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['Subheading'])),(_l()(),i0.ɵted((null as any),
          ['\n                      '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          1,'option',[['selected','']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['Normal'])),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),10,'select',[['class','ql-font']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                      '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),1,'option',[['selected','']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['Sans Serif'])),(_l()(),i0.ɵted((null as any),['\n                      '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'option',[['value','serif']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['Serif'])),(_l()(),i0.ɵted((null as any),['\n                      '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'option',[['value','monospace']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['Monospace'])),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),7,'span',[['class','ql-formats']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Bold'],['class','ql-bold']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Italic'],['class','ql-italic']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Underline'],['class','ql-underline']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),5,'span',[['class','ql-formats']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'select',[['class','ql-color']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'select',[['class','ql-background']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),16,'span',[['class','ql-formats']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Ordered List'],['class','ql-list'],['value','ordered']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'button',[['aria-label','Unordered List'],
              ['class','ql-list'],['value','bullet']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          9,'select',[['class','ql-align']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'option',[['selected','']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'option',[['value','center']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'option',[['value','right']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),0,'option',[['value','justify']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),7,'span',[['class','ql-formats']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Insert Link'],['class','ql-link']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Insert Image'],['class','ql-image']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'button',[['aria-label','Insert Code Block'],['class','ql-code-block']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),3,
          'span',[['class','ql-formats']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'button',[['aria-label','Remove Styles'],
          ['class','ql-clean']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
export function View_Editor_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),11,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,
      (null as any),View_Editor_1)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,
      (null as any),View_Editor_2)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['class',
      'ui-editor-content']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i2.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n        '])),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.Editor = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = 'ui-widget ui-editor-container ui-corner-all';
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.toolbar;
    _ck(_v,5,0,currVal_2);
    const currVal_3:boolean = !_co.toolbar;
    _ck(_v,8,0,currVal_3);
    const currVal_4:any = _co.style;
    _ck(_v,11,0,currVal_4);
  },(null as any));
}
export function View_Editor_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'p-editor',([] as any[]),
      (null as any),(null as any),(null as any),View_Editor_0,RenderType_Editor)),
      i0.ɵprd(5120,(null as any),i4.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i1.Editor]),i0.ɵprd(512,(null as any),i5.DomHandler,i5.DomHandler,([] as any[])),
      i0.ɵdid(4243456,(null as any),1,i1.Editor,[i0.ElementRef,i5.DomHandler],(null as any),
          (null as any)),i0.ɵqud(335544320,1,{toolbar:0})],(null as any),(null as any));
}
export const EditorNgFactory:i0.ComponentFactory<i1.Editor> = i0.ɵccf('p-editor',i1.Editor,
    View_Editor_Host_0,{style:'style',styleClass:'styleClass',placeholder:'placeholder',
        readonly:'readonly',formats:'formats'},{onTextChange:'onTextChange',onSelectionChange:'onSelectionChange',
        onInit:'onInit'},['p-header']);
