// primeng/components/slider/slider.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/slider/slider';
import * as i2 from '@angular/common';
import * as i3 from '@angular/forms';
import * as i4 from 'primeng/components/dom/domhandler';
export const SliderModuleNgFactory:i0.NgModuleFactory<i1.SliderModule> = i0.ɵcmf(i1.SliderModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i1.SliderModule,i1.SliderModule,
          ([] as any[]))]);
    });
const styles_Slider:any[] = ([] as any[]);
export const RenderType_Slider:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Slider,
    data:{}});
function View_Slider_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class',
      'ui-slider-range ui-widget-header ui-corner-all']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgStyle,
      [i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),
      i0.ɵpod({'left':0,width:1})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.handleValues[0] + '%'),((_co.handleValues[1] - _co.handleValues[0]) + '%'));
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_Slider_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class',
      'ui-slider-range ui-widget-header ui-corner-all']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgStyle,
      [i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),
      i0.ɵpod({'bottom':0,height:1})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.handleValues[0] + '%'),((_co.handleValues[1] - _co.handleValues[0]) + '%'));
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_Slider_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class',
      'ui-slider-range ui-slider-range-min ui-widget-header ui-corner-all']],(null as any),
      (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),
      0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
      (null as any)),i0.ɵpod({'height':0})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.handleValue + '%'));
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_Slider_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class',
      'ui-slider-handle ui-state-default ui-corner-all ui-clickable']],[[4,'transition',
      (null as any)]],[[(null as any),'mousedown'],[(null as any),'touchstart'],[(null as any),
      'touchmove'],[(null as any),'touchend']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('mousedown' === en)) {
      const pd_0:any = ((<any>_co.onMouseDown($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('touchstart' === en)) {
      const pd_1:any = ((<any>_co.onTouchStart($event)) !== false);
      ad = (pd_1 && ad);
    }
    if (('touchmove' === en)) {
      const pd_2:any = ((<any>_co.onTouchMove($event)) !== false);
      ad = (pd_2 && ad);
    }
    if (('touchend' === en)) {
      const pd_3:any = ((<any>(_co.dragging = false)) !== false);
      ad = (pd_3 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),i0.ɵpod({'left':0,
      'bottom':1})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = _ck(_v,2,0,((_co.orientation == 'horizontal')? (_co.handleValue + '%'): (null as any)),
        ((_co.orientation == 'vertical')? (_co.handleValue + '%'): (null as any)));
    _ck(_v,1,0,currVal_1);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.dragging? 'none': (null as any));
    _ck(_v,0,0,currVal_0);
  });
}
function View_Slider_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'span',[['class',
      'ui-slider-handle ui-state-default ui-corner-all ui-clickable']],[[4,'transition',
      (null as any)]],[[(null as any),'mousedown'],[(null as any),'touchstart'],[(null as any),
      'touchmove'],[(null as any),'touchend']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('mousedown' === en)) {
      const pd_0:any = ((<any>_co.onMouseDown($event,0)) !== false);
      ad = (pd_0 && ad);
    }
    if (('touchstart' === en)) {
      const pd_1:any = ((<any>_co.onTouchStart($event,0)) !== false);
      ad = (pd_1 && ad);
    }
    if (('touchmove' === en)) {
      const pd_2:any = ((<any>_co.onTouchMove($event,0)) !== false);
      ad = (pd_2 && ad);
    }
    if (('touchend' === en)) {
      const pd_3:any = ((<any>(_co.dragging = false)) !== false);
      ad = (pd_3 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-slider-handle-active':0}),i0.ɵdid(278528,
      (null as any),0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,
          'ngStyle']},(null as any)),i0.ɵpod({'left':0,'bottom':1})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-slider-handle ui-state-default ui-corner-all ui-clickable';
    const currVal_2:any = _ck(_v,2,0,(_co.handleIndex == 0));
    _ck(_v,1,0,currVal_1,currVal_2);
    const currVal_3:any = _ck(_v,4,0,_co.rangeStartLeft,_co.rangeStartBottom);
    _ck(_v,3,0,currVal_3);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.dragging? 'none': (null as any));
    _ck(_v,0,0,currVal_0);
  });
}
function View_Slider_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'span',[['class',
      'ui-slider-handle ui-state-default ui-corner-all ui-clickable']],[[4,'transition',
      (null as any)]],[[(null as any),'mousedown'],[(null as any),'touchstart'],[(null as any),
      'touchmove'],[(null as any),'touchend']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('mousedown' === en)) {
      const pd_0:any = ((<any>_co.onMouseDown($event,1)) !== false);
      ad = (pd_0 && ad);
    }
    if (('touchstart' === en)) {
      const pd_1:any = ((<any>_co.onTouchStart($event,1)) !== false);
      ad = (pd_1 && ad);
    }
    if (('touchmove' === en)) {
      const pd_2:any = ((<any>_co.onTouchMove($event,1)) !== false);
      ad = (pd_2 && ad);
    }
    if (('touchend' === en)) {
      const pd_3:any = ((<any>(_co.dragging = false)) !== false);
      ad = (pd_3 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-slider-handle-active':0}),i0.ɵdid(278528,
      (null as any),0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,
          'ngStyle']},(null as any)),i0.ɵpod({'left':0,'bottom':1})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-slider-handle ui-state-default ui-corner-all ui-clickable';
    const currVal_2:any = _ck(_v,2,0,(_co.handleIndex == 1));
    _ck(_v,1,0,currVal_1,currVal_2);
    const currVal_3:any = _ck(_v,4,0,_co.rangeEndLeft,_co.rangeEndBottom);
    _ck(_v,3,0,currVal_3);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (_co.dragging? 'none': (null as any));
    _ck(_v,0,0,currVal_0);
  });
}
export function View_Slider_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),22,'div',([] as any[]),(null as any),[[(null as any),
          'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.Slider = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.onBarClick($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-slider ui-widget ui-widget-content ui-corner-all':0,
      'ui-state-disabled':1,'ui-slider-horizontal':2,'ui-slider-vertical':3,'ui-slider-animate':4}),
      i0.ɵdid(278528,(null as any),0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,
          i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_Slider_1)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_Slider_2)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_Slider_3)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_Slider_4)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_Slider_5)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_Slider_6)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
          i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.Slider = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = _ck(_v,3,0,true,_co.disabled,(_co.orientation == 'horizontal'),
        (_co.orientation == 'vertical'),_co.animate);
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,4,0,currVal_2);
    const currVal_3:any = (_co.range && (_co.orientation == 'horizontal'));
    _ck(_v,7,0,currVal_3);
    const currVal_4:any = (_co.range && (_co.orientation == 'vertical'));
    _ck(_v,10,0,currVal_4);
    const currVal_5:boolean = (!_co.range && (_co.orientation == 'vertical'));
    _ck(_v,13,0,currVal_5);
    const currVal_6:boolean = !_co.range;
    _ck(_v,16,0,currVal_6);
    const currVal_7:any = _co.range;
    _ck(_v,19,0,currVal_7);
    const currVal_8:any = _co.range;
    _ck(_v,22,0,currVal_8);
  },(null as any));
}
export function View_Slider_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'p-slider',([] as any[]),
      (null as any),(null as any),(null as any),View_Slider_0,RenderType_Slider)),
      i0.ɵprd(5120,(null as any),i3.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i1.Slider]),i0.ɵprd(512,(null as any),i4.DomHandler,i4.DomHandler,([] as any[])),
      i0.ɵdid(180224,(null as any),0,i1.Slider,[i0.ElementRef,i4.DomHandler,i0.Renderer2,
          i0.NgZone],(null as any),(null as any))],(null as any),(null as any));
}
export const SliderNgFactory:i0.ComponentFactory<i1.Slider> = i0.ɵccf('p-slider',i1.Slider,
    View_Slider_Host_0,{animate:'animate',disabled:'disabled',min:'min',max:'max',
        orientation:'orientation',step:'step',range:'range',style:'style',styleClass:'styleClass'},
    {onChange:'onChange',onSlideEnd:'onSlideEnd'},([] as any[]));
