// ng2-archwizard/dist/components/wizard.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from './wizard-navigation-bar.component.ngfactory';
import * as i2 from '@angular/common';
import * as i3 from 'ng2-archwizard/dist/components/wizard-navigation-bar.component';
import * as i4 from 'ng2-archwizard/dist/navigation/wizard-state.model';
import * as i5 from 'ng2-archwizard/dist/components/wizard.component';
const styles_WizardComponent:any[] = ['[_nghost-%COMP%] {\n      display: flex;\n      justify-content: flex-start;\n    }\n    .vertical[_nghost-%COMP%] {\n      flex-direction: row;\n    }\n    .horizontal[_nghost-%COMP%] {\n      flex-direction: column;\n    }\n    [_nghost-%COMP%]   .wizard-steps[_ngcontent-%COMP%] {\n      top: 0;\n      display: flex;\n    }\n    [_nghost-%COMP%]   .wizard-steps.vertical[_ngcontent-%COMP%] {\n      min-width: calc(100% - 280px);\n      width: 80%;\n      height: 100%;\n      flex-direction: column;\n    }\n    [_nghost-%COMP%]   .wizard-steps.horizontal[_ngcontent-%COMP%] {\n      width: 100%;\n      flex-direction: row;\n    }'];
export const RenderType_WizardComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:0,
    styles:styles_WizardComponent,data:{}});
function View_WizardComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'wizard-navigation-bar',
      ([] as any[]),(null as any),(null as any),(null as any),i1.View_WizardNavigationBarComponent_0,
      i1.RenderType_WizardNavigationBarComponent)),i0.ɵdid(278528,(null as any),0,
      i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
      {ngClass:[0,'ngClass']},(null as any)),i0.ɵpod({vertical:0,horizontal:1,small:2,
      'large-filled':3,'large-filled-symbols':4,'large-empty':5,'large-empty-symbols':6}),
      i0.ɵdid(49152,(null as any),0,i3.WizardNavigationBarComponent,[i4.WizardState],
          {direction:[0,'direction']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.navBarLocation == 'left'),(_co.navBarLocation == 'top'),
        (_co.navBarLayout == 'small'),(_co.navBarLayout == 'large-filled'),(_co.navBarLayout == 'large-filled-symbols'),
        (_co.navBarLayout == 'large-empty'),(_co.navBarLayout == 'large-empty-symbols'));
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co.navBarDirection;
    _ck(_v,3,0,currVal_1);
  },(null as any));
}
function View_WizardComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'wizard-navigation-bar',
      ([] as any[]),(null as any),(null as any),(null as any),i1.View_WizardNavigationBarComponent_0,
      i1.RenderType_WizardNavigationBarComponent)),i0.ɵdid(278528,(null as any),0,
      i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
      {ngClass:[0,'ngClass']},(null as any)),i0.ɵpod({vertical:0,horizontal:1,small:2,
      'large-filled':3,'large-filled-symbols':4,'large-empty':5,'large-empty-symbols':6}),
      i0.ɵdid(49152,(null as any),0,i3.WizardNavigationBarComponent,[i4.WizardState],
          {direction:[0,'direction']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.navBarLocation == 'right'),(_co.navBarLocation == 'bottom'),
        (_co.navBarLayout == 'small'),(_co.navBarLayout == 'large-filled'),(_co.navBarLayout == 'large-filled-symbols'),
        (_co.navBarLayout == 'large-empty'),(_co.navBarLayout == 'large-empty-symbols'));
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co.navBarDirection;
    _ck(_v,3,0,currVal_1);
  },(null as any));
}
export function View_WizardComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_WizardComponent_1)),i0.ɵdid(16384,
      (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n\n    '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),5,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,
          'ngClass']},(null as any)),i0.ɵpod({'wizard-steps':0,vertical:1,horizontal:2}),
      (_l()(),i0.ɵted((null as any),['\n      '])),i0.ɵncd((null as any),0),(_l()(),
          i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n\n    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_WizardComponent_2)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i5.WizardComponent = _v.component;
    const currVal_0:any = ((_co.navBarLocation == 'top') || (_co.navBarLocation == 'left'));
    _ck(_v,2,0,currVal_0);
    const currVal_1:any = _ck(_v,6,0,true,((_co.navBarLocation == 'left') || (_co.navBarLocation == 'right')),
        ((_co.navBarLocation == 'top') || (_co.navBarLocation == 'bottom')));
    _ck(_v,5,0,currVal_1);
    const currVal_2:any = ((_co.navBarLocation == 'bottom') || (_co.navBarLocation == 'right'));
    _ck(_v,12,0,currVal_2);
  },(null as any));
}
export function View_WizardComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'wizard',([] as any[]),
      [[2,'horizontal',(null as any)],[2,'vertical',(null as any)]],(null as any),
      (null as any),View_WizardComponent_0,RenderType_WizardComponent)),i0.ɵprd(512,
      (null as any),i4.WizardState,i4.WizardState,([] as any[])),i0.ɵdid(1097728,(null as any),
      1,i5.WizardComponent,[i4.WizardState],(null as any),(null as any)),i0.ɵqud(603979776,
      1,{wizardSteps:1})],(null as any),(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,2).horizontalOrientation;
    const currVal_1:any = i0.ɵnov(_v,2).verticalOrientation;
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
export const WizardComponentNgFactory:i0.ComponentFactory<i5.WizardComponent> = i0.ɵccf('wizard',
    i5.WizardComponent,View_WizardComponent_Host_0,{navBarLocation:'navBarLocation',
        navBarLayout:'navBarLayout',navBarDirection:'navBarDirection',navigationMode:'navigationMode',
        defaultStepIndex:'defaultStepIndex',disableNavigationBar:'disableNavigationBar'},
    {},['*']);
