// primeng/components/multiselect/multiselect.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/multiselect/multiselect';
import * as i2 from '@angular/common';
import * as i3 from 'primeng/components/common/shared';
import * as i4 from '@angular/forms';
import * as i5 from 'primeng/components/dom/domhandler';
import * as i6 from 'primeng/components/utils/objectutils';
export const MultiSelectModuleNgFactory:i0.NgModuleFactory<i1.MultiSelectModule> = i0.ɵcmf(i1.MultiSelectModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i3.SharedModule,i3.SharedModule,
          ([] as any[])),i0.ɵmpd(512,i1.MultiSelectModule,i1.MultiSelectModule,([] as any[]))]);
    });
const styles_MultiSelect:any[] = ([] as any[]);
export const RenderType_MultiSelect:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MultiSelect,
    data:{}});
function View_MultiSelect_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),15,'div',[['class',
      'ui-chkbox ui-widget']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class','ui-helper-hidden-accessible']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵeld(0,
          [['cb',1]],(null as any),0,'input',[['readonly','readonly'],['type','checkbox']],
          [[8,'checked',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵted((null as any),
          ['\n                        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          7,'div',[['class','ui-chkbox-box ui-widget ui-corner-all ui-state-default']],
          (null as any),[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.toggleAll($event,i0.ɵnov(_v,4))) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-state-active':0}),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),2,'span',[['class','ui-chkbox-icon ui-clickable']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'fa fa-check':0}),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵted((null as any),['\n                    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_2:any = _ck(_v,9,0,_co.isAllChecked());
    _ck(_v,8,0,currVal_1,currVal_2);
    const currVal_3:any = 'ui-chkbox-icon ui-clickable';
    const currVal_4:any = _ck(_v,13,0,_co.isAllChecked());
    _ck(_v,12,0,currVal_3,currVal_4);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.isAllChecked();
    _ck(_v,4,0,currVal_0);
  });
}
function View_MultiSelect_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'div',[['class',
      'ui-multiselect-filter-container']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'input',[['class','ui-inputtext ui-widget ui-state-default ui-corner-all'],
          ['role','textbox'],['type','text']],(null as any),[[(null as any),'input']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('input' === en)) {
              const pd_0:any = ((<any>_co.onFilter($event)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','fa fa-fw fa-search']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    ']))],(null as any),(null as any));
}
function View_MultiSelect_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'label',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.label;
    _ck(_v,1,0,currVal_0);
  });
}
function View_MultiSelect_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_MultiSelect_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_MultiSelect_6)),i0.ɵdid(212992,(null as any),0,i3.TemplateWrapper,[i0.ViewContainerRef],
      {item:[0,'item'],index:[1,'index'],templateRef:[2,'templateRef']},(null as any)),
      (_l()(),i0.ɵand(0,(null as any),(null as any),0))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (<any>_v.parent).context.$implicit;
    const currVal_1:any = _co.i;
    const currVal_2:any = _co.itemTemplate;
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
  },(null as any));
}
function View_MultiSelect_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),26,'li',[['class',
      'ui-multiselect-item ui-corner-all']],[[4,'display',(null as any)]],[[(null as any),
      'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.onItemClick($event,_v.context.$implicit.value)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-state-highlight':0}),(_l()(),i0.ɵted((null as any),
      ['\n                            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      15,'div',[['class','ui-chkbox ui-widget']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class','ui-helper-hidden-accessible']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                                    '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),0,'input',[['readonly','readonly'],
              ['type','checkbox']],[[8,'checked',0]],(null as any),(null as any),(null as any),
              (null as any))),(_l()(),i0.ɵted((null as any),['\n                                '])),
      (_l()(),i0.ɵted((null as any),['\n                                '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),7,'div',[['class','ui-chkbox-box ui-widget ui-corner-all ui-state-default']],
              (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'ui-state-active':0}),(_l()(),i0.ɵted((null as any),['\n                                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class','ui-chkbox-icon ui-clickable']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'fa fa-check':0}),(_l()(),i0.ɵted((null as any),['\n                                '])),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵted((null as any),
          ['\n                            '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_MultiSelect_4)),i0.ɵdid(16384,(null as any),
          0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_MultiSelect_5)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                        ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = 'ui-multiselect-item ui-corner-all';
        const currVal_2:any = _ck(_v,2,0,_co.isSelected(_v.context.$implicit.value));
        _ck(_v,1,0,currVal_1,currVal_2);
        const currVal_4:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
        const currVal_5:any = _ck(_v,13,0,_co.isSelected(_v.context.$implicit.value));
        _ck(_v,12,0,currVal_4,currVal_5);
        const currVal_6:any = 'ui-chkbox-icon ui-clickable';
        const currVal_7:any = _ck(_v,17,0,_co.isSelected(_v.context.$implicit.value));
        _ck(_v,16,0,currVal_6,currVal_7);
        const currVal_8:boolean = !_co.itemTemplate;
        _ck(_v,22,0,currVal_8);
        const currVal_9:any = _co.itemTemplate;
        _ck(_v,25,0,currVal_9);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = (_co.isItemVisible(_v.context.$implicit)? 'block': 'none');
        _ck(_v,0,0,currVal_0);
        const currVal_3:any = _co.isSelected(_v.context.$implicit.value);
        _ck(_v,8,0,currVal_3);
      });
}
export function View_MultiSelect_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{containerViewChild:0}),i0.ɵqud(402653184,
      2,{panelViewChild:0}),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
      i0.ɵeld(0,[[1,0],['container',1]],(null as any),53,'div',([] as any[]),(null as any),
          [[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MultiSelect = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.onMouseclick($event,i0.ɵnov(_v,10))) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-multiselect ui-widget ui-state-default ui-corner-all':0,
      'ui-state-focus':1,'ui-state-disabled':2}),i0.ɵdid(278528,(null as any),0,i2.NgStyle,
      [i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),3,'div',[['class','ui-helper-hidden-accessible']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵeld(0,[['in',1]],(null as any),0,'input',
          [['readonly','readonly'],['type','text']],[[1,'id',0],[8,'disabled',0],[1,
              'tabindex',0]],[[(null as any),'focus'],[(null as any),'blur']],(_v,
              en,$event) => {
            var ad:boolean = true;
            var _co:i1.MultiSelect = _v.component;
            if (('focus' === en)) {
              const pd_0:any = ((<any>_co.onFocus($event)) !== false);
              ad = (pd_0 && ad);
            }
            if (('blur' === en)) {
              const pd_1:any = ((<any>_co.onInputBlur($event)) !== false);
              ad = (pd_1 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),4,'div',[['class','ui-multiselect-label-container']],[[8,'title',
              0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),
          i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'label',[['class','ui-multiselect-label ui-corner-all']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['',''])),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),5,'div',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,
              'ngClass']},(null as any)),i0.ɵpod({'ui-multiselect-trigger ui-state-default ui-corner-right':0}),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,'span',[['class','fa fa-fw fa-caret-down ui-clickable']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵeld(0,[[2,0],['panel',1]],(null as any),
          29,'div',([] as any[]),[[4,'display',(null as any)]],[[(null as any),'click']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.MultiSelect = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>(_co.panelClick = true)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,
              'ngClass']},(null as any)),i0.ɵpad(2),i0.ɵdid(278528,(null as any),0,
          i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),14,'div',[['class','ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix']],
              (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'ui-multiselect-header-no-toggleall':0}),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MultiSelect_1)),i0.ɵdid(16384,(null as any),0,i2.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_MultiSelect_2)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'a',[['class','ui-multiselect-close ui-corner-all'],
          ['href','#']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.MultiSelect = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.close($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','fa fa-close']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),7,'div',[['class','ui-multiselect-items-wrapper']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),4,'ul',[['class','ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset']],
          [[4,'max-height',(null as any)]],(null as any),(null as any),(null as any),
          (null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MultiSelect_3)),
      i0.ɵdid(802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.MultiSelect = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = _ck(_v,5,0,true,_co.focus,_co.disabled);
    _ck(_v,4,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,6,0,currVal_2);
    const currVal_8:any = _ck(_v,21,0,true);
    _ck(_v,20,0,currVal_8);
    const currVal_10:any = _ck(_v,28,0,'ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow',
        (_co.panelStyleClass || ''));
    _ck(_v,27,0,currVal_10);
    const currVal_11:any = _co.panelStyle;
    _ck(_v,29,0,currVal_11);
    const currVal_12:any = 'ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix';
    const currVal_13:any = _ck(_v,33,0,!_co.showToggleAll);
    _ck(_v,32,0,currVal_12,currVal_13);
    const currVal_14:any = _co.showToggleAll;
    _ck(_v,36,0,currVal_14);
    const currVal_15:any = _co.filter;
    _ck(_v,39,0,currVal_15);
    const currVal_17:any = _co.options;
    _ck(_v,52,0,currVal_17);
  },(_ck,_v) => {
    var _co:i1.MultiSelect = _v.component;
    const currVal_3:any = _co.inputId;
    const currVal_4:any = _co.disabled;
    const currVal_5:any = _co.tabindex;
    _ck(_v,10,0,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = _co.valuesAsString;
    _ck(_v,13,0,currVal_6);
    const currVal_7:any = _co.valuesAsString;
    _ck(_v,16,0,currVal_7);
    const currVal_9:any = (_co.overlayVisible? 'block': 'none');
    _ck(_v,26,0,currVal_9);
    const currVal_16:any = (_co.scrollHeight || 'auto');
    _ck(_v,49,0,currVal_16);
  });
}
export function View_MultiSelect_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'p-multiSelect',
      ([] as any[]),(null as any),(null as any),(null as any),View_MultiSelect_0,RenderType_MultiSelect)),
      i0.ɵprd(5120,(null as any),i4.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i1.MultiSelect]),i0.ɵprd(512,(null as any),i5.DomHandler,i5.DomHandler,([] as any[])),
      i0.ɵprd(512,(null as any),i6.ObjectUtils,i6.ObjectUtils,([] as any[])),i0.ɵdid(14139392,
          (null as any),1,i1.MultiSelect,[i0.ElementRef,i5.DomHandler,i0.Renderer2,
              i0.IterableDiffers,i6.ObjectUtils,i0.ChangeDetectorRef],(null as any),
          (null as any)),i0.ɵqud(603979776,1,{templates:1})],(_ck,_v) => {
    _ck(_v,4,0);
  },(null as any));
}
export const MultiSelectNgFactory:i0.ComponentFactory<i1.MultiSelect> = i0.ɵccf('p-multiSelect',
    i1.MultiSelect,View_MultiSelect_Host_0,{options:'options',scrollHeight:'scrollHeight',
        defaultLabel:'defaultLabel',style:'style',styleClass:'styleClass',panelStyle:'panelStyle',
        panelStyleClass:'panelStyleClass',inputId:'inputId',disabled:'disabled',filter:'filter',
        overlayVisible:'overlayVisible',tabindex:'tabindex',appendTo:'appendTo',dataKey:'dataKey',
        displaySelectedLabel:'displaySelectedLabel',maxSelectedLabels:'maxSelectedLabels',
        selectedItemsLabel:'selectedItemsLabel',showToggleAll:'showToggleAll'},{onChange:'onChange',
        onBlur:'onBlur'},([] as any[]));
