// primeng/components/checkbox/checkbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/checkbox/checkbox';
import * as i2 from '@angular/common';
import * as i3 from '@angular/forms';
export const CheckboxModuleNgFactory:i0.NgModuleFactory<i1.CheckboxModule> = i0.ɵcmf(i1.CheckboxModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i1.CheckboxModule,i1.CheckboxModule,
          ([] as any[]))]);
    });
const styles_Checkbox:any[] = ([] as any[]);
export const RenderType_Checkbox:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Checkbox,
    data:{}});
function View_Checkbox_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'label',[['class',
      'ui-chkbox-label']],[[1,'for',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.onClick($event,i0.ɵnov((<any>_v.parent),7),true)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-label-active':0,'ui-label-disabled':1,
      'ui-label-focus':2}),(_l()(),i0.ɵted((null as any),['','']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-chkbox-label';
    const currVal_2:any = _ck(_v,2,0,_co.checked,_co.disabled,_co.focused);
    _ck(_v,1,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.inputId;
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = _co.label;
    _ck(_v,3,0,currVal_3);
  });
}
export function View_Checkbox_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),19,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵdid(278528,(null as any),
      0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),5,'div',[['class','ui-helper-hidden-accessible']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,[['cb',1]],
      (null as any),2,'input',[['type','checkbox']],[[1,'id',0],[8,'name',0],[8,'value',
          0],[8,'checked',0],[8,'disabled',0],[1,'tabindex',0]],[[(null as any),'focus'],
          [(null as any),'blur'],[(null as any),'change']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i1.Checkbox = _v.component;
        if (('focus' === en)) {
          const pd_0:any = ((<any>_co.onFocus($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('blur' === en)) {
          const pd_1:any = ((<any>_co.onBlur($event)) !== false);
          ad = (pd_1 && ad);
        }
        if (('change' === en)) {
          const pd_2:any = ((<any>_co.handleChange($event)) !== false);
          ad = (pd_2 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,'ngClass']},(null as any)),
      i0.ɵpod({'ui-state-focus':0}),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),7,'div',[['class','ui-chkbox-box ui-widget ui-corner-all ui-state-default']],
          (null as any),[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i1.Checkbox = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.onClick($event,i0.ɵnov(_v,7),true)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-state-active':0,
          'ui-state-disabled':1,'ui-state-focus':2}),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),2,
          'span',[['class','ui-chkbox-icon ui-clickable']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),
          0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
          {klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'fa fa-check':0}),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Checkbox_1)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,
      _v) => {
    var _co:i1.Checkbox = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = 'ui-chkbox ui-widget';
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,3,0,currVal_2);
    const currVal_9:any = _ck(_v,9,0,_co.focused);
    _ck(_v,8,0,currVal_9);
    const currVal_10:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_11:any = _ck(_v,14,0,_co.checked,_co.disabled,_co.focused);
    _ck(_v,13,0,currVal_10,currVal_11);
    const currVal_12:any = 'ui-chkbox-icon ui-clickable';
    const currVal_13:any = _ck(_v,18,0,_co.checked);
    _ck(_v,17,0,currVal_12,currVal_13);
    const currVal_14:any = _co.label;
    _ck(_v,23,0,currVal_14);
  },(_ck,_v) => {
    var _co:i1.Checkbox = _v.component;
    const currVal_3:any = _co.inputId;
    const currVal_4:any = _co.name;
    const currVal_5:any = _co.value;
    const currVal_6:any = _co.checked;
    const currVal_7:any = _co.disabled;
    const currVal_8:any = _co.tabindex;
    _ck(_v,7,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
export function View_Checkbox_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'p-checkbox',([] as any[]),
      (null as any),(null as any),(null as any),View_Checkbox_0,RenderType_Checkbox)),
      i0.ɵprd(5120,(null as any),i3.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i1.Checkbox]),i0.ɵdid(49152,(null as any),0,i1.Checkbox,[i0.ChangeDetectorRef],
          (null as any),(null as any))],(null as any),(null as any));
}
export const CheckboxNgFactory:i0.ComponentFactory<i1.Checkbox> = i0.ɵccf('p-checkbox',
    i1.Checkbox,View_Checkbox_Host_0,{value:'value',name:'name',disabled:'disabled',
        binary:'binary',label:'label',tabindex:'tabindex',inputId:'inputId',style:'style',
        styleClass:'styleClass'},{onChange:'onChange'},([] as any[]));
