// primeng/components/organizationchart/organizationchart.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/organizationchart/organizationchart';
import * as i2 from '@angular/common';
import * as i3 from 'primeng/components/common/shared';
import * as i4 from 'primeng/components/dom/domhandler';
export const OrganizationChartModuleNgFactory:i0.NgModuleFactory<i1.OrganizationChartModule> = i0.ɵcmf(i1.OrganizationChartModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i3.SharedModule,i3.SharedModule,
          ([] as any[])),i0.ɵmpd(512,i1.OrganizationChartModule,i1.OrganizationChartModule,
          ([] as any[]))]);
    });
const styles_OrganizationChart:any[] = ([] as any[]);
export const RenderType_OrganizationChart:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_OrganizationChart,data:{}});
function View_OrganizationChart_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'table',[['class',
      'ui-organizationchart-table'],['pOrganizationChartNode','']],(null as any),(null as any),
      (null as any),View_OrganizationChartNode_0,RenderType_OrganizationChartNode)),
      i0.ɵdid(49152,(null as any),0,i1.OrganizationChartNode,[i1.OrganizationChart],
          {node:[0,'node']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.root;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
export function View_OrganizationChart_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵdid(278528,(null as any),
      0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_OrganizationChart_1)),i0.ɵdid(16384,
      (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),
      ['\n    ']))],(_ck,_v) => {
    var _co:i1.OrganizationChart = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = 'ui-organizationchart ui-widget';
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,3,0,currVal_2);
    const currVal_3:any = _co.root;
    _ck(_v,6,0,currVal_3);
  },(null as any));
}
export function View_OrganizationChart_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'p-organizationChart',
      ([] as any[]),(null as any),(null as any),(null as any),View_OrganizationChart_0,
      RenderType_OrganizationChart)),i0.ɵprd(512,(null as any),i4.DomHandler,i4.DomHandler,
      ([] as any[])),i0.ɵdid(1097728,(null as any),1,i1.OrganizationChart,[i0.ElementRef,
      i4.DomHandler],(null as any),(null as any)),i0.ɵqud(603979776,1,{templates:1})],
      (null as any),(null as any));
}
export const OrganizationChartNgFactory:i0.ComponentFactory<i1.OrganizationChart> = i0.ɵccf('p-organizationChart',
    i1.OrganizationChart,View_OrganizationChart_Host_0,{value:'value',style:'style',
        styleClass:'styleClass',selectionMode:'selectionMode',selection:'selection'},
    {selectionChange:'selectionChange',onNodeSelect:'onNodeSelect',onNodeUnselect:'onNodeUnselect'},
    ([] as any[]));
const styles_OrganizationChartNode:any[] = ([] as any[]);
export const RenderType_OrganizationChartNode:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_OrganizationChartNode,data:{'animation':[{type:7,name:'childState',
        definitions:[{type:0,name:'in',styles:{type:6,styles:{opacity:1},offset:(null as any)}},
            {type:1,expr:'void => *',animation:[{type:6,styles:{opacity:0},offset:(null as any)},
                {type:4,styles:(null as any),timings:150}],options:(null as any)},
            {type:1,expr:'* => void',animation:[{type:4,styles:{type:6,styles:{opacity:0},
                offset:(null as any)},timings:150}],options:(null as any)}],options:{}}]}});
function View_OrganizationChartNode_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.node.label;
    _ck(_v,1,0,currVal_0);
  });
}
function View_OrganizationChartNode_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵeld(16777216,
      (null as any),(null as any),1,'p-organizationChartNodeTemplateLoader',([] as any[]),
      (null as any),(null as any),(null as any),View_OrganizationChartNodeTemplateLoader_0,
      RenderType_OrganizationChartNodeTemplateLoader)),i0.ɵdid(245760,(null as any),
      0,i1.OrganizationChartNodeTemplateLoader,[i0.ViewContainerRef],{node:[0,'node'],
          template:[1,'template']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                    ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.node;
        const currVal_1:any = _co.chart.getTemplateForNode(_co.node);
        _ck(_v,3,0,currVal_0,currVal_1);
      },(null as any));
}
function View_OrganizationChartNode_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'a',[['class',
      'ui-node-toggler'],['href','#']],(null as any),[[(null as any),'click']],(_v,
      en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.toggleNode($event,_co.node)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'i',[['class','fa ui-node-toggler-icon']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'fa-chevron-down':0,'fa-chevron-up':1}),(_l()(),i0.ɵted((null as any),
          ['\n                    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = 'fa ui-node-toggler-icon';
    const currVal_1:any = _ck(_v,4,0,_co.node.expanded,!_co.node.expanded);
    _ck(_v,3,0,currVal_0,currVal_1);
  },(null as any));
}
function View_OrganizationChartNode_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),18,'tr',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      15,'td',([] as any[]),[[1,'colspan',0]],(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),12,'div',([] as any[]),(null as any),[[(null as any),
          'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.onNodeClick($event,_co.node)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-organizationchart-selectable-node':0,
      'ui-state-highlight':1}),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_2)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_3)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_OrganizationChartNode_4)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = i0.ɵinlineInterpolate(1,'ui-organizationchart-node-content ui-widget-content ui-corner-all ',
        _co.node.styleClass,'');
    const currVal_2:any = _ck(_v,6,0,(_co.chart.selectionMode && (_co.node.selectable !== false)),
        _co.isSelected());
    _ck(_v,5,0,currVal_1,currVal_2);
    const currVal_3:boolean = !_co.chart.getTemplateForNode(_co.node);
    _ck(_v,9,0,currVal_3);
    const currVal_4:any = _co.chart.getTemplateForNode(_co.node);
    _ck(_v,12,0,currVal_4);
    const currVal_5:boolean = !_co.leaf;
    _ck(_v,15,0,currVal_5);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.colspan;
    _ck(_v,2,0,currVal_0);
  });
}
function View_OrganizationChartNode_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),3,'td',[['class','ui-organizationchart-line-left']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'ui-organizationchart-line-top':0}),(_l()(),i0.ɵted((null as any),[' '])),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),3,'td',[['class','ui-organizationchart-line-right']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,
          (null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,
              i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'ui-organizationchart-line-top':0}),(_l()(),i0.ɵted((null as any),[' '])),
      (_l()(),i0.ɵted((null as any),['\n            ']))],(_ck,_v) => {
    const currVal_0:any = 'ui-organizationchart-line-left';
    const currVal_1:any = _ck(_v,3,0,!_v.context.first);
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = 'ui-organizationchart-line-right';
    const currVal_3:any = _ck(_v,8,0,!_v.context.last);
    _ck(_v,7,0,currVal_2,currVal_3);
  },(null as any));
}
function View_OrganizationChartNode_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'td',[['colspan',
      '2']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'table',[['class','ui-organizationchart-table'],['pOrganizationChartNode',
              '']],(null as any),(null as any),(null as any),View_OrganizationChartNode_0,
          RenderType_OrganizationChartNode)),i0.ɵdid(49152,(null as any),0,i1.OrganizationChartNode,
          [i1.OrganizationChart],{node:[0,'node']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            ']))],(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
export function View_OrganizationChartNode_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_OrganizationChartNode_1)),i0.ɵdid(16384,
      (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
      (null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),6,'tr',[['class','ui-organizationchart-lines']],
      [[4,'visibility',(null as any)],[24,'@childState',0]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'td',([] as any[]),[[1,'colspan',
          0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),0,
          'div',[['class','ui-organizationchart-line-down']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          4,'tr',[['class','ui-organizationchart-lines']],[[4,'visibility',(null as any)],
              [24,'@childState',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_OrganizationChartNode_5)),
      i0.ɵdid(802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),4,'tr',[['class','ui-organizationchart-nodes']],
              [[4,'visibility',(null as any)],[24,'@childState',0]],(null as any),
              (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_OrganizationChartNode_6)),i0.ɵdid(802816,(null as any),
          0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
              'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.OrganizationChartNode = _v.component;
    const currVal_0:any = _co.node;
    _ck(_v,2,0,currVal_0);
    const currVal_6:any = _co.node.children;
    _ck(_v,15,0,currVal_6);
    const currVal_9:any = _co.node.children;
    _ck(_v,21,0,currVal_9);
  },(_ck,_v) => {
    var _co:i1.OrganizationChartNode = _v.component;
    const currVal_1:any = ((!_co.leaf && _co.node.expanded)? 'inherit': 'hidden');
    const currVal_2:any = 'in';
    _ck(_v,4,0,currVal_1,currVal_2);
    const currVal_3:any = _co.colspan;
    _ck(_v,6,0,currVal_3);
    const currVal_4:any = ((!_co.leaf && _co.node.expanded)? 'inherit': 'hidden');
    const currVal_5:any = 'in';
    _ck(_v,12,0,currVal_4,currVal_5);
    const currVal_7:any = ((!_co.leaf && _co.node.expanded)? 'inherit': 'hidden');
    const currVal_8:any = 'in';
    _ck(_v,18,0,currVal_7,currVal_8);
  });
}
export function View_OrganizationChartNode_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',[['pOrganizationChartNode',
      '']],(null as any),(null as any),(null as any),View_OrganizationChartNode_0,
      RenderType_OrganizationChartNode)),i0.ɵdid(49152,(null as any),0,i1.OrganizationChartNode,
      [i1.OrganizationChart],(null as any),(null as any))],(null as any),(null as any));
}
export const OrganizationChartNodeNgFactory:i0.ComponentFactory<i1.OrganizationChartNode> = i0.ɵccf('[pOrganizationChartNode]',
    i1.OrganizationChartNode,View_OrganizationChartNode_Host_0,{node:'node',root:'root',
        first:'first',last:'last'},{},([] as any[]));
const styles_OrganizationChartNodeTemplateLoader:any[] = ([] as any[]);
export const RenderType_OrganizationChartNodeTemplateLoader:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_OrganizationChartNodeTemplateLoader,data:{}});
export function View_OrganizationChartNodeTemplateLoader_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
export function View_OrganizationChartNodeTemplateLoader_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),1,'p-organizationChartNodeTemplateLoader',
      ([] as any[]),(null as any),(null as any),(null as any),View_OrganizationChartNodeTemplateLoader_0,
      RenderType_OrganizationChartNodeTemplateLoader)),i0.ɵdid(245760,(null as any),
      0,i1.OrganizationChartNodeTemplateLoader,[i0.ViewContainerRef],(null as any),
      (null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const OrganizationChartNodeTemplateLoaderNgFactory:i0.ComponentFactory<i1.OrganizationChartNodeTemplateLoader> = i0.ɵccf('p-organizationChartNodeTemplateLoader',
    i1.OrganizationChartNodeTemplateLoader,View_OrganizationChartNodeTemplateLoader_Host_0,
    {node:'node',template:'template'},{},([] as any[]));
