// primeng/components/megamenu/megamenu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/megamenu/megamenu';
import * as i2 from '@angular/common';
import * as i3 from '@angular/router';
import * as i4 from 'primeng/components/dom/domhandler';
export const MegaMenuModuleNgFactory:i0.NgModuleFactory<i1.MegaMenuModule> = i0.ɵcmf(i1.MegaMenuModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i3.RouterModule,i3.RouterModule,
          [[2,i3.ɵa],[2,i3.Router]]),i0.ɵmpd(512,i1.MegaMenuModule,i1.MegaMenuModule,
          ([] as any[]))]);
    });
const styles_MegaMenu:any[] = ([] as any[]);
export const RenderType_MegaMenu:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_MegaMenu,
    data:{}});
function View_MegaMenu_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'li',[['class',
      'ui-menu-separator ui-widget-content']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                    ']))],
      (null as any),(null as any));
}
function View_MegaMenu_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'li',[['class',
      'ui-menu-separator ui-widget-content']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                                                    ']))],
      (null as any),(null as any));
}
function View_MegaMenu_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'ui-menuitem-icon fa fa-fw']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any))],(_ck,_v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>_v.parent).parent).parent).context.$implicit.icon;
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_MegaMenu_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'a',[['class',
      'ui-menuitem-link ui-corner-all']],[[8,'href',4],[1,'target',0],[1,'title',0]],
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.itemClick($event,(<any>(<any>_v.parent).parent).context.$implicit)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-state-disabled':0}),(_l()(),i0.ɵted((null as any),
      ['\n                                                            '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_10)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                                                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class','ui-menuitem-text']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['',''])),(_l()(),i0.ɵted((null as any),['\n                                                        ']))],
      (_ck,_v) => {
        const currVal_3:any = 'ui-menuitem-link ui-corner-all';
        const currVal_4:any = _ck(_v,2,0,(<any>(<any>_v.parent).parent).context.$implicit.disabled);
        _ck(_v,1,0,currVal_3,currVal_4);
        const currVal_5:any = (<any>(<any>_v.parent).parent).context.$implicit.icon;
        _ck(_v,5,0,currVal_5);
      },(_ck,_v) => {
        const currVal_0:any = ((<any>(<any>_v.parent).parent).context.$implicit.url || '#');
        const currVal_1:any = (<any>(<any>_v.parent).parent).context.$implicit.target;
        const currVal_2:any = (<any>(<any>_v.parent).parent).context.$implicit.title;
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
        const currVal_6:any = (<any>(<any>_v.parent).parent).context.$implicit.label;
        _ck(_v,8,0,currVal_6);
      });
}
function View_MegaMenu_12(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'ui-menuitem-icon fa fa-fw']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any))],(_ck,_v) => {
    const currVal_0:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_1:any = (<any>(<any>(<any>_v.parent).parent).parent).context.$implicit.icon;
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_MegaMenu_11(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),14,'a',[['class',
      'ui-menuitem-link ui-corner-all']],[[1,'target',0],[1,'title',0],[1,'target',
      0],[8,'href',4]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,3).onClick($event.button,$event.ctrlKey,$event.metaKey,
          $event.shiftKey)) !== false);
      ad = (pd_0 && ad);
    }
    if (('click' === en)) {
      const pd_1:any = ((<any>_co.itemClick($event,(<any>(<any>_v.parent).parent).context.$implicit)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-state-disabled':0}),i0.ɵdid(671744,[[2,
      4]],0,i3.RouterLinkWithHref,[i3.Router,i3.ActivatedRoute,i2.LocationStrategy],
      {routerLink:[0,'routerLink']},(null as any)),i0.ɵdid(1720320,(null as any),2,
      i3.RouterLinkActive,[i3.Router,i0.ElementRef,i0.Renderer,i0.ChangeDetectorRef],
      {routerLinkActiveOptions:[0,'routerLinkActiveOptions'],routerLinkActive:[1,'routerLinkActive']},
      (null as any)),i0.ɵqud(603979776,1,{links:1}),i0.ɵqud(603979776,2,{linksWithHrefs:1}),
      i0.ɵpod({exact:0}),(_l()(),i0.ɵted((null as any),['\n                                                            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_12)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                                                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class','ui-menuitem-text']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['',''])),(_l()(),i0.ɵted((null as any),['\n                                                        ']))],
      (_ck,_v) => {
        const currVal_4:any = 'ui-menuitem-link ui-corner-all';
        const currVal_5:any = _ck(_v,2,0,(<any>(<any>_v.parent).parent).context.$implicit.disabled);
        _ck(_v,1,0,currVal_4,currVal_5);
        const currVal_6:any = (<any>(<any>_v.parent).parent).context.$implicit.routerLink;
        _ck(_v,3,0,currVal_6);
        const currVal_7:any = ((<any>(<any>_v.parent).parent).context.$implicit.routerLinkActiveOptions || _ck(_v,
            7,0,false));
        const currVal_8:any = 'ui-state-active';
        _ck(_v,4,0,currVal_7,currVal_8);
        const currVal_9:any = (<any>(<any>_v.parent).parent).context.$implicit.icon;
        _ck(_v,10,0,currVal_9);
      },(_ck,_v) => {
        const currVal_0:any = (<any>(<any>_v.parent).parent).context.$implicit.target;
        const currVal_1:any = (<any>(<any>_v.parent).parent).context.$implicit.title;
        const currVal_2:any = i0.ɵnov(_v,3).target;
        const currVal_3:any = i0.ɵnov(_v,3).href;
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
        const currVal_10:any = (<any>(<any>_v.parent).parent).context.$implicit.label;
        _ck(_v,13,0,currVal_10);
      });
}
function View_MegaMenu_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'li',[['class',
      'ui-menuitem ui-widget ui-corner-all']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                                                        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_9)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                                                        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_11)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                                                    ']))],
      (_ck,_v) => {
        const currVal_0:boolean = !(<any>_v.parent).context.$implicit.routerLink;
        _ck(_v,3,0,currVal_0);
        const currVal_1:any = (<any>_v.parent).context.$implicit.routerLink;
        _ck(_v,6,0,currVal_1);
      },(null as any));
}
function View_MegaMenu_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n                                                    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_7)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MegaMenu_8)),i0.ɵdid(16384,(null as any),0,i2.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n                                                ']))],
      (_ck,_v) => {
        const currVal_0:any = _v.context.$implicit.separator;
        _ck(_v,2,0,currVal_0);
        const currVal_1:boolean = !_v.context.$implicit.separator;
        _ck(_v,4,0,currVal_1);
      },(null as any));
}
function View_MegaMenu_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n                                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),8,'ul',[['class','ui-menu-list ui-helper-reset']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                                                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'li',[['class','ui-widget-header ui-corner-all']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'h3',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['',''])),(_l()(),i0.ɵted((null as any),['\n                                                '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_6)),
      i0.ɵdid(802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n                                            '])),(_l()(),i0.ɵted((null as any),
          ['\n                                        ']))],(_ck,_v) => {
    const currVal_1:any = _v.context.$implicit.items;
    _ck(_v,8,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.label;
    _ck(_v,5,0,currVal_0);
  });
}
function View_MegaMenu_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n                                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),[[8,'className',
          0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                                        '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_MegaMenu_5)),i0.ɵdid(802816,
          (null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                                    '])),
      (_l()(),i0.ɵted((null as any),['\n                                ']))],(_ck,
      _v) => {
    const currVal_1:any = _v.context.$implicit;
    _ck(_v,4,0,currVal_1);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.getColumnClass((<any>(<any>_v.parent).parent).context.$implicit);
    _ck(_v,1,0,currVal_0);
  });
}
function View_MegaMenu_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,[['item',1]],(null as any),27,'li',([] as any[]),
      (null as any),[[(null as any),'mouseenter'],[(null as any),'mouseleave']],(_v,
          en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('mouseenter' === en)) {
          const pd_0:any = ((<any>_co.onItemMouseEnter($event,i0.ɵnov(_v,0),(<any>_v.parent).context.$implicit)) !== false);
          ad = (pd_0 && ad);
        }
        if (('mouseleave' === en)) {
          const pd_1:any = ((<any>_co.onItemMouseLeave($event,i0.ɵnov(_v,0))) !== false);
          ad = (pd_1 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,'ngClass']},(null as any)),
      i0.ɵpod({'ui-menuitem ui-widget ui-corner-all':0,'ui-menu-parent':1,'ui-menuitem-active':2}),
      (_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),13,'a',[['class','ui-menuitem-link ui-corner-all ui-submenu-link']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'ui-state-disabled':0}),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class','ui-menuitem-icon fa fa-fw']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),1,'span',[['class','ui-menuitem-text']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['',''])),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class','ui-submenu-icon fa fa-fw']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'fa-caret-down':0,'fa-caret-right':1}),(_l()(),i0.ɵted((null as any),
          ['\n                        '])),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),7,'div',[['class','ui-megamenu-panel ui-widget-content ui-menu-list ui-corner-all ui-helper-clearfix ui-menu-child ui-shadow']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),4,'div',[['class','ui-g']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                                '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_MegaMenu_4)),i0.ɵdid(802816,(null as any),
          0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
              'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵted((null as any),
          ['\n                    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,true,(<any>_v.parent).context.$implicit.items,
        (i0.ɵnov(_v,0) == _co.activeItem));
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = 'ui-menuitem-link ui-corner-all ui-submenu-link';
    const currVal_2:any = _ck(_v,6,0,(<any>_v.parent).context.$implicit.disabled);
    _ck(_v,5,0,currVal_1,currVal_2);
    const currVal_3:any = 'ui-menuitem-icon fa fa-fw';
    const currVal_4:any = (<any>_v.parent).context.$implicit.icon;
    _ck(_v,9,0,currVal_3,currVal_4);
    const currVal_6:any = 'ui-submenu-icon fa fa-fw';
    const currVal_7:any = _ck(_v,16,0,(_co.orientation == 'horizontal'),(_co.orientation == 'vertical'));
    _ck(_v,15,0,currVal_6,currVal_7);
    const currVal_8:any = (<any>_v.parent).context.$implicit.items;
    _ck(_v,24,0,currVal_8);
  },(_ck,_v) => {
    const currVal_5:any = (<any>_v.parent).context.$implicit.label;
    _ck(_v,12,0,currVal_5);
  });
}
function View_MegaMenu_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_2)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MegaMenu_3)),i0.ɵdid(16384,(null as any),0,i2.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n                ']))],(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.separator;
    _ck(_v,2,0,currVal_0);
    const currVal_1:boolean = !_v.context.$implicit.separator;
    _ck(_v,4,0,currVal_1);
  },(null as any));
}
function View_MegaMenu_13(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'li',[['class',
      'ui-menuitem ui-menuitem-custom ui-widget ui-corner-all']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),
      i0.ɵncd((null as any),0),(_l()(),i0.ɵted((null as any),['\n                ']))],
      (null as any),(null as any));
}
export function View_MegaMenu_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),13,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-menu ui-menubar ui-megamenu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix':0,
      'ui-megamenu-vertical':1}),i0.ɵdid(278528,(null as any),0,i2.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'ul',[['class',
      'ui-menu-list ui-helper-reset ui-menubar-root-list']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_MegaMenu_1)),
      i0.ɵdid(802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_MegaMenu_13)),i0.ɵdid(16384,(null as any),0,i2.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i1.MegaMenu = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = _ck(_v,3,0,true,(_co.orientation == 'vertical'));
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,4,0,currVal_2);
    const currVal_3:any = _co.model;
    _ck(_v,9,0,currVal_3);
    const currVal_4:any = (_co.orientation === 'horizontal');
    _ck(_v,12,0,currVal_4);
  },(null as any));
}
export function View_MegaMenu_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'p-megaMenu',([] as any[]),
      (null as any),(null as any),(null as any),View_MegaMenu_0,RenderType_MegaMenu)),
      i0.ɵprd(512,(null as any),i4.DomHandler,i4.DomHandler,([] as any[])),i0.ɵdid(49152,
          (null as any),0,i1.MegaMenu,[i0.ElementRef,i4.DomHandler,i0.Renderer2],(null as any),
          (null as any))],(null as any),(null as any));
}
export const MegaMenuNgFactory:i0.ComponentFactory<i1.MegaMenu> = i0.ɵccf('p-megaMenu',
    i1.MegaMenu,View_MegaMenu_Host_0,{model:'model',style:'style',styleClass:'styleClass',
        orientation:'orientation'},{},['*']);
