// primeng/components/listbox/listbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from 'primeng/components/listbox/listbox';
import * as i2 from '@angular/common';
import * as i3 from 'primeng/components/common/shared';
import * as i4 from '@angular/forms';
import * as i5 from 'primeng/components/dom/domhandler';
import * as i6 from 'primeng/components/utils/objectutils';
export const ListboxModuleNgFactory:i0.NgModuleFactory<i1.ListboxModule> = i0.ɵcmf(i1.ListboxModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(4608,
          i2.NgLocalization,i2.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.CommonModule,
          i2.CommonModule,([] as any[])),i0.ɵmpd(512,i3.SharedModule,i3.SharedModule,
          ([] as any[])),i0.ɵmpd(512,i1.ListboxModule,i1.ListboxModule,([] as any[]))]);
    });
const styles_Listbox:any[] = ([] as any[]);
export const RenderType_Listbox:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_Listbox,
    data:{}});
function View_Listbox_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),15,'div',[['class',
      'ui-chkbox ui-widget']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),3,'div',[['class','ui-helper-hidden-accessible']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵeld(0,
          [['cb',1]],(null as any),0,'input',[['readonly','readonly'],['type','checkbox']],
          [[8,'checked',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          7,'div',[['class','ui-chkbox-box ui-widget ui-corner-all ui-state-default']],
          (null as any),[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.toggleAll($event,i0.ɵnov(_v,4))) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
          [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
              'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-state-active':0}),
      (_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),2,'span',[['class','ui-chkbox-icon ui-clickable']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'fa fa-check':0}),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵted((null as any),['\n                ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_2:any = _ck(_v,9,0,_co.allChecked);
    _ck(_v,8,0,currVal_1,currVal_2);
    const currVal_3:any = 'ui-chkbox-icon ui-clickable';
    const currVal_4:any = _ck(_v,13,0,_co.allChecked);
    _ck(_v,12,0,currVal_3,currVal_4);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.allChecked;
    _ck(_v,4,0,currVal_0);
  });
}
function View_Listbox_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'div',[['class',
      'ui-listbox-filter-container']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),0,'input',[['class','ui-inputtext ui-widget ui-state-default ui-corner-all'],
          ['role','textbox'],['type','text']],[[8,'disabled',0]],[[(null as any),'input']],
          (_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('input' === en)) {
              const pd_0:any = ((<any>_co.onFilter($event)) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class','fa fa-search']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.disabled;
    _ck(_v,2,0,currVal_0);
  });
}
function View_Listbox_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'ui-widget-header ui-corner-all ui-listbox-header ui-helper-clearfix']],(null as any),
      (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),
      0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],
      {klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-listbox-header-w-checkbox':0}),
      (_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_Listbox_2)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Listbox_3)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = 'ui-widget-header ui-corner-all ui-listbox-header ui-helper-clearfix';
        const currVal_1:any = _ck(_v,2,0,_co.checkbox);
        _ck(_v,1,0,currVal_0,currVal_1);
        const currVal_2:any = ((_co.checkbox && _co.multiple) && _co.showToggleAll);
        _ck(_v,5,0,currVal_2);
        const currVal_3:any = _co.filter;
        _ck(_v,8,0,currVal_3);
      },(null as any));
}
function View_Listbox_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),15,'div',[['class',
      'ui-chkbox ui-widget']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.onCheckboxClick($event,(<any>_v.parent).context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class','ui-helper-hidden-accessible']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n                                '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),0,'input',[['type','checkbox']],[[8,
              'checked',0],[8,'disabled',0]],(null as any),(null as any),(null as any),
              (null as any))),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),7,'div',[['class','ui-chkbox-box ui-widget ui-corner-all ui-state-default']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'ui-state-active':0}),(_l()(),i0.ɵted((null as any),['\n                                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class','ui-chkbox-icon ui-clickable']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,
          i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'fa fa-check':0}),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵted((null as any),['\n                        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
    const currVal_3:any = _ck(_v,9,0,_co.isSelected((<any>_v.parent).context.$implicit));
    _ck(_v,8,0,currVal_2,currVal_3);
    const currVal_4:any = 'ui-chkbox-icon ui-clickable';
    const currVal_5:any = _ck(_v,13,0,_co.isSelected((<any>_v.parent).context.$implicit));
    _ck(_v,12,0,currVal_4,currVal_5);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.isSelected((<any>_v.parent).context.$implicit);
    const currVal_1:any = _co.disabled;
    _ck(_v,4,0,currVal_0,currVal_1);
  });
}
function View_Listbox_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.label;
    _ck(_v,1,0,currVal_0);
  });
}
function View_Listbox_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_Listbox_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_Listbox_8)),i0.ɵdid(212992,(null as any),0,i3.TemplateWrapper,[i0.ViewContainerRef],
      {item:[0,'item'],index:[1,'index'],templateRef:[2,'templateRef']},(null as any)),
      (_l()(),i0.ɵand(0,(null as any),(null as any),0))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (<any>_v.parent).context.$implicit;
    const currVal_1:any = (<any>_v.parent).context.index;
    const currVal_2:any = _co.itemTemplate;
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
  },(null as any));
}
function View_Listbox_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),12,'li',([] as any[]),
      [[4,'display',(null as any)]],[[(null as any),'click'],[(null as any),'dblclick'],
          [(null as any),'touchend']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.onOptionClick($event,_v.context.$implicit)) !== false);
          ad = (pd_0 && ad);
        }
        if (('dblclick' === en)) {
          const pd_1:any = ((<any>_co.onDoubleClick($event,_v.context.$implicit)) !== false);
          ad = (pd_1 && ad);
        }
        if (('touchend' === en)) {
          const pd_2:any = ((<any>_co.onOptionTouchEnd($event,_v.context.$implicit)) !== false);
          ad = (pd_2 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,'ngClass']},(null as any)),
      i0.ɵpod({'ui-listbox-item ui-corner-all':0,'ui-state-highlight':1}),(_l()(),
          i0.ɵted((null as any),['\n                        '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_Listbox_5)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Listbox_6)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Listbox_7)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                    ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = _ck(_v,2,0,true,_co.isSelected(_v.context.$implicit));
        _ck(_v,1,0,currVal_1);
        const currVal_2:any = (_co.checkbox && _co.multiple);
        _ck(_v,5,0,currVal_2);
        const currVal_3:boolean = !_co.itemTemplate;
        _ck(_v,8,0,currVal_3);
        const currVal_4:any = _co.itemTemplate;
        _ck(_v,11,0,currVal_4);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = (_co.isItemVisible(_v.context.$implicit)? 'block': 'none');
        _ck(_v,0,0,currVal_0);
      });
}
function View_Listbox_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'div',[['class',
      'ui-listbox-footer ui-widget-header ui-corner-all']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),
      i0.ɵncd((null as any),0),(_l()(),i0.ɵted((null as any),['\n            ']))],
      (null as any),(null as any));
}
export function View_Listbox_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),20,'div',([] as any[]),(null as any),(null as any),
      (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i2.NgClass,
      [i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,
          'klass'],ngClass:[1,'ngClass']},(null as any)),i0.ɵpod({'ui-listbox ui-inputtext ui-widget ui-widget-content ui-corner-all':0,
      'ui-state-disabled':1}),i0.ɵdid(278528,(null as any),0,i2.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,
      (null as any),View_Listbox_1)),i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),8,'div',[['class',
      'ui-listbox-list-wrapper']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),5,'ul',[['class','ui-listbox-list']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,
      (null as any),0,i2.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,
          'ngStyle']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Listbox_4)),
      i0.ɵdid(802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_Listbox_9)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),
          ['\n    ']))],(_ck,_v) => {
    var _co:i1.Listbox = _v.component;
    const currVal_0:any = _co.styleClass;
    const currVal_1:any = _ck(_v,3,0,true,_co.disabled);
    _ck(_v,2,0,currVal_0,currVal_1);
    const currVal_2:any = _co.style;
    _ck(_v,4,0,currVal_2);
    const currVal_3:any = ((_co.checkbox && _co.multiple) || _co.filter);
    _ck(_v,7,0,currVal_3);
    const currVal_4:any = _co.listStyle;
    _ck(_v,12,0,currVal_4);
    const currVal_5:any = _co.options;
    _ck(_v,15,0,currVal_5);
    const currVal_6:any = _co.footerFacet;
    _ck(_v,20,0,currVal_6);
  },(null as any));
}
export function View_Listbox_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),6,'p-listbox',([] as any[]),
      (null as any),(null as any),(null as any),View_Listbox_0,RenderType_Listbox)),
      i0.ɵprd(5120,(null as any),i4.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i1.Listbox]),i0.ɵprd(512,(null as any),i5.DomHandler,i5.DomHandler,([] as any[])),
      i0.ɵprd(512,(null as any),i6.ObjectUtils,i6.ObjectUtils,([] as any[])),i0.ɵdid(1097728,
          (null as any),2,i1.Listbox,[i0.ElementRef,i5.DomHandler,i6.ObjectUtils],
          (null as any),(null as any)),i0.ɵqud(335544320,1,{footerFacet:0}),i0.ɵqud(603979776,
          2,{templates:1})],(null as any),(null as any));
}
export const ListboxNgFactory:i0.ComponentFactory<i1.Listbox> = i0.ɵccf('p-listbox',
    i1.Listbox,View_Listbox_Host_0,{options:'options',multiple:'multiple',style:'style',
        styleClass:'styleClass',listStyle:'listStyle',disabled:'disabled',checkbox:'checkbox',
        filter:'filter',metaKeySelection:'metaKeySelection',dataKey:'dataKey',showToggleAll:'showToggleAll'},
    {onChange:'onChange',onDblClick:'onDblClick'},['p-footer']);
